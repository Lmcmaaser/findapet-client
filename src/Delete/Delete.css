.delete-form {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 20px;
  background-color: #AD264C;
  border: 2px solid #AD264C;
  border-radius: 30px;
  margin-left: 10px;
  margin-right: 10px;
}

h2 {
  text-align: center;
  color: white;
}

.section {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  font-size: 18px;
  padding-top: 0.35em;
  padding-bottom: 0.35em;
  padding-left: 0.75em;
  padding-right: 0.75em;
  color: white;
}

.pet-section {
  justify-content: center;
  margin: 20px;
  font-size: 20px;
  padding: 20px;
  border: 3px solid white;
  border-radius: 10px;
}

button.submit-button {
  border-radius: 20px;
  border: 2px solid white;
  display: block;
  font-size: 16px;
  font-weight: bold;
  height: 70px;
  padding: 8px;
  text-align: center;
  width: 90px;
  margin: 20px 20px 20px auto;
}

@media screen and (max-width: 530px) {
  .section {
    flex-direction: column;
  }

  .pet-section {
    margin: 10px;
  }
}
