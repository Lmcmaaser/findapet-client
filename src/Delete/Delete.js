import React from 'react';
import ApiContext from '../ApiContext';
import config from '../config'
import './Delete.css'

class Delete extends React.Component {
  static contextType = ApiContext;

  updateStatus(remove) {
    this.setState({remove: {value: remove, touched: true}});
    console.log(remove) //shows input
  }

  handleSubmit(event, pet) {
    event.preventDefault();
    fetch(`${config.API_ENDPOINT}/delete/${id}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${config.API_TOKEN}`
      }
    })
      .then(res => {
        if (!res.ok) {
          return res.json().then(event => Promise.reject(event))
        }
        return res
      })
      .then((deletePet) => {
        this.context.deletePet(pet);
        this.props.history.push('/');
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const { pets=[] } = this.context;
    const pet = pets.find(pet => pet.id === this.props.match.params.id)
    return (
      <div className="delete-form" >
        <h2>Would you like to delete this pet?</h2>
          <div className="section">
            <div className="pet-section">
              <div key={pet.id}>
                <div>Name: {pet.name}<br /></div>
                <div>Type: {pet.pet_type}<br /></div>
                <div>Sex: {pet.sex}<br /></div>
                <div>Age: {pet.age}<br /></div>
                <div>Adopted: {pet.adopted}<br /></div>
              </div>
            </div>
              <button
                className="submit-button"
                type="submit"
                onClick={(event) => {this.handleSubmit(event, pet)
                }}
              >
                Delete
              </button>
              <button
                className="submit-button"
                onClick={() => {
                  this.props.history.push('/')
                }}
              >
                Cancel
              </button>
          </div>
      </div>
    )
  }
}

export default Delete;
