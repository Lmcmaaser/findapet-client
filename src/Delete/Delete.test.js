import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Delete from './Delete'

it('renders a .delete-form by default', () => {
  const wrapper = shallow(<Delete />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders without crashing', () => {
  shallow(<Delete />);
});
