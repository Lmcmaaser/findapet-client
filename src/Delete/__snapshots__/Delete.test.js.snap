// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a .delete-form by default 1`] = `<Fragment />`;
